File: Loom.Core/Models/MemoryContext.vb
Namespace Models
    ''' <summary>
    ''' Memory state of single session
    ''' </summary>
    Public Class MemoryContext
        Public Property Mode As Enums.MemoryMode = Enums.MemoryMode.FullHistory
        Public Property Content As String
        Public Property TTL As TimeSpan? ' How long memory must persist
    End Class
End Namespace

File: implementations/csharp/MemoryContext.cs
using System;

namespace Models
{
    /// <summary>
    /// Memory state of single session
    /// </summary>
    public class MemoryContext
    {
        public Enums.MemoryMode Mode { get; set; } = Enums.MemoryMode.FullHistory;
        public string Content { get; set; }
        public TimeSpan? TTL { get; set; } // How long memory must persist
    }
}